### PR TITLE
Xenobio Additions and Tweaks

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Just a few additions to Xenobio, to make it slightly more useful to the station and have a more interesting end-game---not to mention making an existing reaction not totally worthless.
- Adds in support for self-recharging cells; they regain 25% of their recharge rate every `process` (this is 75% slower than putting them in a charger)
- Slime cells self-recharge by default
- Added in Slime transference potion; this potion is similar to sentience potions---the difference is, however, is that it puts the _user_ in control of the simple animal instead of pulling from the ghost population. The user is given the notification that they are still of the same allegiance, alliance, and role as they were prior to transference. Can obtain this potion by injecting a rainbow slime core with blood.

:cl: Fox McCloud
add: Slime batteries now self-recharge.
add: Adds transference potion. A potion that allows the user to transfer their consciousness to a simple mob.
add: Adds rainbow slime core reaction that generates a consciousness transference potion when injected with blood.
/:cl:

### CODE REVIEW

**Comment by ChangelingRain** [comment]
I'd like to note here that rezadone heals all clone damage each tick and this should probably give as little as three units because it can be diluted with water or split with a chemmaster for 'heal all clone damage' very easily.

**Comment by Fox-McCloud** [comment]
>I'd like to note here that rezadone heals all clone damage each tick and this should probably give as little as three units because it can be diluted with water or split with a chemmaster for 'heal all clone damage' very easily.

Sure. though you can also just spawn carp and butcher them for effectively the same thing---Still, I'll keep it in mind and make changes come tomorrow.

**Comment by Iamgoofball** [comment]
>I'd like to note here that rezadone heals all clone damage each tick

I THOUGHT I KILLED THE LAST OF THESE FUCKERS WITH DEXPLUS

BRB

**Comment by Fox-McCloud** [comment]
@Iamgoofball 

Well, the problem was, Rezadone healed about the same/less damage than cryox---so literally not one could be assed to use it, so....it was buffed to remove all cloneloss---and it's still not used...mostly because it requires a random event or xenobio to acquire.

**Comment by jughu** [comment]
doesnt require a random event just mutate soybeans into koibeans they include carpotoxin i used to grow them for greed and he makes a ton of rezadone

**Comment by KorPhaeron** [comment]
Giving absolutely no healing chems to xenobio was a very deliberate choice. They need to not be totally self sufficient.

They should be visiting medbay if they get hurt.

**Comment by Fox-McCloud** [comment]
>Giving absolutely no healing chems to xenobio was a very deliberate choice. They need to not be totally self sufficient.

Maybe so, but R&D and Botany are both in the "100% self sufficient" category already, by virture of hte wide products they can produce (and R&D's chem dispenser). 

Also, there's literally no risk involved in Xenobio with the camera console (or feeding each slime a single sheet at round start to make them and all their offspring your permanent friends).

**Comment by Fox-McCloud** [comment]
@ChangelingRain 

I lowered the rezadone amount, as you requested.

**Comment by Cheridan** [comment]
>They should be visiting medbay if they get hurt.

seconding this

**Comment by Fox-McCloud** [comment]
Legitmate question for @Cheridan and @KorPhaeron 

This may sound snarky, and I'm not trying to, it's just a legit question.

Why is it ok for R&D to be self-sufficient and self healing at ~20 minutes in (chem dispensers), but Xenobio, which would have to invest a lot more resources and time to get just a single healing chem is not ok?

**Comment by KorPhaeron** [comment]
>Why is it ok for R&D to be self-sufficient and self healing at ~20 minutes in

It's not, it's the result of endless unchecked creep in RD

**Comment by Fox-McCloud** [comment]
@KorPhaeron 

Alright, reverted the healing chems---just the transference+cells now

**Comment by coiax** [comment]
This should get merged, rainbow slimes are tough to get anyway. Self recharging cells are also an actually interesting way of R&D doing something sciencey to help with power. Could make a SMES filled with them.

Although, in case no one has thought of it, the most powerful use of the transference potion is to transfer yourself into a headcrab. Then that'll reopen all the fun traitor + changeling memes. But you've probably earned it if you've spent 30 minutes breeding slimes and ~~then have to get a dead body~~ then bite your old body to death and lay an egg in it and wait six minutes.

But :+1:

**Comment by Fox-McCloud** [comment]
@ChangelingRain @KorPhaeron 

This acceptable to merge now that the healing chems have been removed?

**Comment by KorPhaeron** [comment]
yeah, just waiting on code review

File: code/modules/power/cell.dm [hunk 1]
```diff
@@ -17,14 +17,26 @@
 	var/rigged = 0		// true if rigged to explode
 	var/minor_fault = 0 //If not 100% reliable, it will build up faults.
 	var/chargerate = 100 //how much power is given every tick in a recharger
+	var/self_recharge = 0 //does it self recharge, over time, or not?
```
Comment (ChangelingRain) [inline code/modules/power/cell.dm]: You should do what energy guns do with on_varedit() so admins can edit this var to have a cell recharge.
https://github.com/tgstation/-tg-station/blob/master/code/modules/projectiles/guns/energy.dm#L160

Comment (Fox-McCloud) [inline code/modules/power/cell.dm]: Oh. Thanks. I didn't know that was a thing; that problem occurred to me when putting this in, but I didn't want to have non-charging cells in object processing, so I opted for this--didn't realize that there was a proc like this. Will implement.

**Comment by Fox-McCloud** [comment]
Alright @ChangelingRain 

Added in the `on_varedit` for cells.

**Pull request merged**